src/test/java/htsjdk/samtools/BAMFileWriterTest.java

```diff
@@ -342,7 +339,7 @@ public void testBinNotNullWhenLargeCigarIsLoaded(final int numOps) throws Except
         try (final SamReader reader = SamReaderFactory.makeDefault().validationStringency(ValidationStringency.SILENT).open(bamFile)) {
             reader.iterator().forEachRemaining(samRecord -> {
                 samRecord.getCigar();
-                Assert.assertNotNull(samRecord.getIndexingBin());
+                samRecord.computeIndexingBin();
```

Comment:
you removed the assertion...what does this check now?

Reply:
(add a comment if there might be a thrown exception that you are checking for)

Reply:
I have no real idea what it was checking before - the only thing it can have been checking is that a value got returned.  Since we don't store it any more but calculate it on demand, the return type of `computeIndexingBin()` is `int`, so null checking didn't seem appropriate.  But the method will throw an exception if it can't be calculated for some reason.  So it's testing that it can be generated.  Much like the `samRecord.getCigar()` directly above it that someone else (ahem) wrote ;)

Reply:
The way it's written now the test will pass even if `computeIndexingBin()` is never called, e.g. if `reader.iterator()` returns an empty stream. Is there a way to improve this?

> the method will throw an exception if it can't be calculated for some reason

Are there any negative tests that ensure that this will happen?